xbmc/pictures/Picture.cpp

```diff
@@ -284,18 +284,19 @@ std::unique_ptr<CTexture> CPicture::CreateTiledThumb(const std::vector<std::stri
   if (!files.size())
     return {};
 
-  unsigned int num_across = (unsigned int)ceil(sqrt((float)files.size()));
+  unsigned int num_across =
+      static_cast<unsigned int>(std::ceil(std::sqrt(static_cast<float>(files.size()))));
   unsigned int num_down = (files.size() + num_across - 1) / num_across;
 
   unsigned int imageRes = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_imageRes;
 
   unsigned int tile_width = imageRes / num_across;
   unsigned int tile_height = imageRes / num_down;
   unsigned int tile_gap = 1;
-  bool success = false;
+  bool success = false; // Flag that we at least had one successful image processed
 
   // create a buffer for the resulting thumb
-  uint32_t* buffer = static_cast<uint32_t*>(calloc(imageRes * imageRes, 4));
+  std::unique_ptr<uint32_t[]> buffer(new uint32_t[imageRes * imageRes]);
```

Comment:
Good job with this change. The only other step would be to use `std::make_unique`. That way, the pointer is hidden entirely, which should be our goal.